xarray/coding/variables.py

```diff
@@ -345,7 +345,14 @@ def encode(self, variable: Variable, name: T_Name = None):
         if fill_value is not None and has_unsigned:
             pop_to(encoding, attrs, "_Unsigned")
             # XXX: Is this actually needed? Doesn't the backend handle this?
-            data = duck_array_ops.astype(duck_array_ops.around(data), dtype)
+            signed_dtype = np.dtype(f"i{data.itemsize}")
```

Comment:
@dcherian @Illviljan Does this agree with your suggestion? This first casts the rounded float to int (of same itemsize) and in a second step the int to the final intN (where N is the wanted itemsize).

@QuLogic Does this work on your machine type?

Reply:
Yes, this passes tests on all architectures.